### PR TITLE
allow routing between small components if path exists

### PR DESCRIPTION
**Pull request by MoKob**
# Issue

First step on the way to https://github.com/Project-OSRM/osrm-backend/issues/3443. This PR only handles direct shortest paths and does not handle via-routes.

## Tasklist
 - [x] add regression / cucumber cases (see docs/testing.md)
 - [ ] review
 - [ ] adjust for comments

## Requirements / Relations
 none

### CODE REVIEW

File: src/engine/plugins/viaroute.cpp [hunk 1]
```diff
@@ -82,21 +82,46 @@ Status ViaRoutePlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataF
                 !continue_straight_at_waypoint;
         }
     };
-    util::for_each_pair(snapped_phantoms, build_phantom_pairs);
 
-    if (1 == raw_route.segment_end_coordinates.size())
+    if (phantom_node_pairs.size() == 2)
```
Comment (oxidase) [inline src/engine/plugins/viaroute.cpp]: :+1: for removing Yoda style

Comment (MoKob) [inline src/engine/plugins/viaroute.cpp]: Yeah. kind of doesn't help next to `const` functions. Can save you from `if( var = 2 )` errors, though.

Comment (daniel-j-h) [inline src/engine/plugins/viaroute.cpp]: I think this is on by default (`-Wall`):

`-Wparentheses`
> Warn if parentheses are omitted in certain contexts, such as when there is an assignment in a context where a truth value is expected, or when operators are nested whose precedence people often get confused about.

**Comment by TheMarex** [comment]
The main problem I see here is consistency in snapping behavior. Sure this will work for two endpoints but how does this generalize to three or four? How does this work with distance matrices? Keeping the discussion in #3443.

I think we should wait until we have a solution that generalizes before moving on changing behavior in one plugin only.

**Pull request closed without merging**